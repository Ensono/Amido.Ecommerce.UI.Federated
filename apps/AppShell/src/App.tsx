/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import {FC, StrictMode, createContext} from 'react'
import {Link, Route, Routes} from 'react-router-dom'

import {Text} from '@batman-ui-components/text'
import {context, federateComponent} from '@batman/federate-component'
import {getRemoteUrls} from '@batman/remote-urls'

import logo from './logo.svg'
import './App.css'

// TODO: what is this?
export {context}

const ThemeContext = createContext(null)
const ReduxContext = createContext(null)

export const ThemeProvider = ({children, data}: any) => {
  return <ThemeContext.Provider value={data}>{children}</ThemeContext.Provider>
}

export const ReduxProvider = ({children, data}: any) => {
  return <ReduxContext.Provider value={data}>{children}</ReduxContext.Provider>
}

const remotesUrls = getRemoteUrls()

const Header = federateComponent('mfe_header', './header', remotesUrls.mfe_header)
const Footer = federateComponent('mfe_footer', './footer', remotesUrls.mfe_footer)
const ProductListing = federateComponent('mfe_product_listing', './product-listing', remotesUrls.mfe_product_listing)
const ProductDetails = federateComponent('mfe_product_details', './product-details', remotesUrls.mfe_product_details)

/**
 * Host app that consumes micro-front-ends and renders them together
 * @remarks
 *
 */
const App: FC = () => {
  return (
    <StrictMode>
      <ThemeProvider value={{}}>
        <Header loadingFallback={<div>Loading header...</div>} errorFallback={<div>Error loading header</div>}>
          SSR header!
          <Link to="/app">Home</Link>
          <Link to="/app/productListing">Product Listing</Link>
        </Header>
        <Routes>
          <Route
            path="/app"
            element={
              <div className="App">
                <section className="App-header">
                  <img src={logo} className="App-logo" alt="logo" />
                  {/* eslint-disable-next-line no-console */}
                  <p onClick={() => console.log('lallero')}>Page 1</p>
                  <Text />
                </section>
              </div>
            }
          />
          <Route
            path="/app/productListing"
            element={
              <ProductListing
                loadingFallback={<div>Loading product listing...</div>}
                errorFallback={<div>Error loading product listing</div>}
              />
            }
          />
          <Route
            path="/app/productDetails/:id"
            element={
              <ProductDetails
                loadingFallback={<div>Loading product details...</div>}
                errorFallback={<div>Error loading product details</div>}
              />
            }
          />
        </Routes>
        <Footer loadingFallback={<div>Loading footer...</div>} errorFallback={<div>Error loading footer</div>}>
          SSR footer!
        </Footer>
      </ThemeProvider>
    </StrictMode>
  )
}

export default App
