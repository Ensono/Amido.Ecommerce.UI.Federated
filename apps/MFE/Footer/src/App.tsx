import {StrictMode, createContext} from 'react'

import {ThemeProvider} from '@batman/contexts'

import Footer from './Footer'

const ReduxContext = createContext(null)

export const ReduxProvider = ({children, data}: any) => {
  return <ReduxContext.Provider value={data}>{children}</ReduxContext.Provider>
}

/**
 * Renders the Footer MFE as a standalone app.
 * Not used by consumers of the Footer as a federated module.
 */
const App: React.FC = () => {
  return (
    <StrictMode>
      <ThemeProvider>
        <Footer />
      </ThemeProvider>
    </StrictMode>
  )
}

export default App
