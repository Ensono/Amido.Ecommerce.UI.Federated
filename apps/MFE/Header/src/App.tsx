import {StrictMode, createContext} from 'react'

import {Text} from '@batman-ui-components/text'

import Header from './Header'

const ThemeContext = createContext(null)

export const ThemeProvider = ({children, data}: any) => {
  return <ThemeContext.Provider value={data}>{children}</ThemeContext.Provider>
}
 // test
/**
 * Renders the Header MFE in a standalone page.
 * Not used by consumers of the Header federated module.
 */
const App: React.FC = () => {
  return (
    <StrictMode>
      <ThemeProvider value={{}}>
        <Header>
          This is the header children with a text component
          <Text />
        </Header>
      </ThemeProvider>
    </StrictMode>
  )
}

export default App
