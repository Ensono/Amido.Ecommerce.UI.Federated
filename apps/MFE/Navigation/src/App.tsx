import {StrictMode, createContext} from 'react'

import {Text} from '@batman-ui-components/text'

import Navigation from './Navigation'

const ThemeContext = createContext(null)
const ReduxContext = createContext(null)

export const ThemeProvider = ({children, data}: any) => {
  return <ThemeContext.Provider value={data}>{children}</ThemeContext.Provider>
}

export const ReduxProvider = ({children, data}: any) => {
  return <ReduxContext.Provider value={data}>{children}</ReduxContext.Provider>
}

/**
 * Renders the Navigation MFE as a standalone app.
 * Not used by consumers of the Navigation federated module.
 */
const App: React.FC = () => {
  return (
    <StrictMode>
      <ThemeProvider value={{}}>
        <Navigation position="top" />
        <Text />
        <Navigation position="bottom" />
      </ThemeProvider>
    </StrictMode>
  )
}

export default App
