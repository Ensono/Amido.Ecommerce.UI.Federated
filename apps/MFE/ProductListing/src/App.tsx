import {StrictMode, createContext} from 'react'

import ProductListing from './ProductListing'

const ThemeContext = createContext(null)
const ReduxContext = createContext(null)

export const ThemeProvider = ({children, data}: any) => {
  return <ThemeContext.Provider value={data}>{children}</ThemeContext.Provider>
}

export const ReduxProvider = ({children, data}: any) => {
  return <ReduxContext.Provider value={data}>{children}</ReduxContext.Provider>
}

/**
 * Renders the ProductListing MFE in a standalone page.
 * Not used by consumers of the ProductListing federated module.
 */
const App: React.FC = () => {
  return (
    <StrictMode>
      <ThemeProvider value={{}}>
        <ProductListing />
      </ThemeProvider>
    </StrictMode>
  )
}

export default App
