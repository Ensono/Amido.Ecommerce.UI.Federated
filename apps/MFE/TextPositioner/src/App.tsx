import {StrictMode, createContext} from 'react'

import {Text} from '@batman-ui-components/text'
import {ThemeProvider} from '@batman/contexts'

import TextPositioner from './TextPositioner'

const ReduxContext = createContext(null)

export const ReduxProvider = ({children, data}: any) => {
  return <ReduxContext.Provider value={data}>{children}</ReduxContext.Provider>
}

/**
 * Renders the TextPositioner MFE as a standalone app.
 * Not used by consumers of the TextPositioner federated module.
 */
const App: React.FC = () => {
  return (
    <StrictMode>
      <ThemeProvider>
        <TextPositioner position="top">Positioned top by text positioner</TextPositioner>
        <Text />
        <TextPositioner position="bottom">Positioned bottom by text positioner</TextPositioner>
      </ThemeProvider>
    </StrictMode>
  )
}

export default App
