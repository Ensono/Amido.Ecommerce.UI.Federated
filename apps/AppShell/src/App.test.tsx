import React from 'react'

import { render, screen } from '@testing-library/react'

import App from './App.tsx'

test('renders external components', () => {
  render(<App />)
  const linkElement = screen.getByText(/This comes from UI! 1.0.0/i)
  const sharedTextComponent = screen.getByText(/Shared text/i)

  expect(linkElement).toBeInTheDocument()
  expect(sharedTextComponent).toBeInTheDocument()
})
