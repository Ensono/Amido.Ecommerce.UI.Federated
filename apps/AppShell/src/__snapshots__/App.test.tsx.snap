// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Routes should naviagate successfully to the /app page 1`] = `
<div>
  <section>
    mfe_header
  </section>
  <div
    class="App"
  >
    <section
      class="App-header"
    >
      <p>
        Page 1
      </p>
      <div>
        <h2
          data-testid="count"
        >
          Count: 
          <span>
            0
          </span>
        </h2>
        <button
          data-testid="increment-button"
        >
          Make number go up plz
        </button>
        <button
          data-testid="reset-button"
        >
          Reset
        </button>
      </div>
      Shared text
    </section>
  </div>
  <section>
    mfe_footer
  </section>
</div>
`;

exports[`App Routes should naviagate successfully to the /app/error-example page 1`] = `
<div>
  <section>
    mfe_header
  </section>
  <section>
    mfe_footer
  </section>
</div>
`;

exports[`App Routes should naviagate successfully to the /app/productDetails/1 page 1`] = `
<div>
  <section>
    mfe_header
  </section>
  <section>
    mfe_product_details
  </section>
  <section>
    mfe_footer
  </section>
</div>
`;

exports[`App Routes should naviagate successfully to the /app/productListing page 1`] = `
<div>
  <section>
    mfe_header
  </section>
  <section>
    mfe_product_listing
  </section>
  <section>
    mfe_footer
  </section>
</div>
`;
